### ISSUE TITLE
[expo-notifications] Unable to instantiate service ExpoNotificationsService: ClassNotFoundException

### ISSUE THREAD

**Issue by RodolfoGS** [expo/expo#10618]
### Summary of Issue 

Hi, I'm getting this crash on Crashlytics (Android). Seems that `ExpoNotificationsService` doesn't exist and it's right, doesn't exist. This class is used by: https://github.com/expo/expo/blob/ef941d8a564379cc3b14a7078cabf16d0be836f9/packages/expo-notifications/android/src/main/AndroidManifest.xml#L15 

But doesn't exist in this folder:
https://github.com/expo/expo/tree/master/packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/service

![image](https://user-images.githubusercontent.com/870365/95582770-b81b3280-0a00-11eb-8ea3-04c94043c29a.png)

### Environment - output of `expo diagnostics` & the platform(s) you're targeting

```
  Expo CLI 3.27.14 environment info:
    System:
      OS: macOS 10.15.5
      Shell: 5.7.1 - /bin/zsh
    Binaries:
      Node: 14.13.0 - /usr/local/bin/node
      Yarn: 1.19.1 - /usr/local/bin/yarn
      npm: 6.14.8 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    Managers:
      CocoaPods: 1.9.3 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: iOS 14.0, DriverKit 19.0, macOS 10.15, tvOS 14.0, watchOS 7.0
      Android SDK:
        Android NDK: 21.3.6528147
    IDEs:
      Android Studio: 4.0 AI-193.6911.18.40.6514223
      Xcode: 12.0/12A7209 - /usr/bin/xcodebuild
    npmPackages:
      expo: ~39.0.3 => 39.0.3
      react: 16.13.1 => 16.13.1
      react-native: ~0.63.3 => 0.63.3
      expo-notifications: 0.7.2
    npmGlobalPackages:
      expo-cli: 3.27.14
    Expo Workflow: bare
```

### Reproducible Demo

I don't know how to reproduce, in my device works great. But I've the crash in Crashlytics and reading the code seems that this class doesn't exist.
 
### Steps to Reproduce

Open `node_modules/expo-notifications/android/src/main/AndroidManifest.xml` and see `.notifications.service.ExpoNotificationsService` in red.

### Expected Behavior vs Actual Behavior

**Expected Behavior**: crash
**Actual Behavior**: no crash

**Comment by cruzach** [comment]
Yeah it looks like that class was removed in [this commit](https://github.com/expo/expo/commit/d40fb974f9c6310011e69e0bcf375725362c2458). @mczernek are we okay to remove the remaining references to `ExpoNotificationsService`? Looks like it's just declared in the `AndroidManifest` files and in some comments
